src/ResourceManager/HDInsight/Commands.HDInsight/ManagementCommands/AddAzureHDInsightSecurityProfile.cs

```diff
@@ -0,0 +1,98 @@
+using System.Linq;
+using Microsoft.Azure.Commands.HDInsight.Commands;
+using Microsoft.Azure.Commands.HDInsight.Models;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.HDInsight
+{
+    [Cmdlet(
+        VerbsCommon.Add,
+        Constants.CommandNames.AzureHDInsightSecurityProfile),
+    OutputType(
+        typeof(AzureHDInsightSecurityProfile))]
+    public class AddAzureHDInsightSecurityProfile : HDInsightCmdletBase
+    {
+        private readonly AzureHDInsightSecurityProfile _securityProfile;
+
+        #region Input Parameter Definitions
+
+        [Parameter(Position = 0,
+        Mandatory = true,
+        ValueFromPipeline = true,
+        HelpMessage = "The HDInsight cluster configuration to use when creating the new cluster.")]
+        public AzureHDInsightConfig Config { get; set; }
+
+        [Parameter(Position = 1,
+            Mandatory = true,
+            HelpMessage = "Active Directory domain for the cluster.")]
+        public string Domain
+        {
+            get { return _securityProfile.Domain; }
+            set { _securityProfile.Domain = value; } 
+        }
+
+        [Parameter(Position = 2,
+            Mandatory = true,
+            HelpMessage = "A domain user account credential with sufficient permissions for creating the cluster. Username should be in user@domain format.")]
+        public PSCredential DomainUserCredential 
+        {
+            get { return _securityProfile.DomainUserCredential; }
+            set { _securityProfile.DomainUserCredential = value; } 
+        }
+
+        [Parameter(Position = 3,
+            Mandatory = true,
+            HelpMessage = "Distinguished name of the organizational unit in the Active directory where user and computer accounts will be created.")]
+        public string OrganizationalUnitDN
+        {
+            get { return _securityProfile.OrganizationalUnitDN; }
+            set { _securityProfile.OrganizationalUnitDN = value; }
+        }
+
+        [Parameter(Position = 4,
+            Mandatory = true,
+            HelpMessage = "Urls of one or multiple LDAPS servers for the Active Directory.")]
+        public string[] LdapsUrls
+        {
+            get { return _securityProfile.LdapsUrls.ToArray(); }
+            set { _securityProfile.LdapsUrls = value; }
+        }
+
+        [Parameter(Position = 5,
+            HelpMessage = "Distinguished names of the Active Directory groups that will be available in Ambari and Ranger.")]
+        public string[] ClusterUsersGroupDNs
+        {
+            get { return _securityProfile.ClusterUsersGroupDNs.ToArray(); }
+            set { _securityProfile.ClusterUsersGroupDNs = value; }
+        }
+
+        #endregion
+
+        public AddAzureHDInsightSecurityProfile()
+        {
+            _securityProfile = new AzureHDInsightSecurityProfile();
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            Config.SecurityProfile = _securityProfile;
```

Comment:
Here we are not adding anything to azure and no network call is made. It is used as a helper cmdlet to add security profile configuration object to the Config object. This config object is later used in New-AzureRmHDInsightCluster to create an HDInsight cluster. This operation is safe and should not require permission from user.